lib/active_fedora/change_set.rb

```diff
@@ -19,8 +19,8 @@ def empty?
 
     def changes
       @changes ||= changed_attributes.each_with_object({}) do |key, result|
-        if m = /^.*_ids?$/.match(key)
-          predicate = object.association(m[0].to_sym).reflection.predicate
+        if object.respond_to?(:association) && object.association(key.to_sym).present?
```

Comment:
There was a test failing when changes was being called on a File which doesn't respond to association.

https://github.com/projecthydra/active_fedora/blob/master/spec/integration/with_metadata_spec.rb#L43